### PR TITLE
Reorganize hook methods

### PR DESCRIPTION
**Pull request by vvmruder**
There are 2 targets with this PR:

1. organize code to fit better in the new pattern of contrib and core concept
2. improve test ability via separating concerns of methods where it belongs

The decrease of coverage comes from code reorganization I think. Code which was accidentally called through some methods is dissolved now and is not "tested"...

It will be tested in dedicated future tests.

### CODE REVIEW

**Comment by vvmruder** [comment]
The decrease of coverage comes from code reorganization I think. Now code which was accidentally called through some methods is now dissolved and will be tested in dedicated tests.

File: pyramid_oereb/core/views/webservice.py [hunk 1]
```diff
@@ -803,17 +802,28 @@ def get_image(self):
         Returns:
             pyramid.response.Response: Response containing the binary image content.
         """
-        method = None
-        dnr = DottedNameResolver()
-        for plr in Config.get('plrs'):
-            if str(plr.get('code')).lower() == str(self._request_.matchdict.get('theme_code')).lower():
-                method = dnr.resolve(plr.get('hooks').get('get_symbol'))
-                break
+        theme_code = self._request_.matchdict.get('theme_code')
```
Comment (jwkaltz) [inline pyramid_oereb/core/views/webservice.py:805]: Why are we changing webservice.py in this pull request? Is it necessary for the hook method reorganization? If not, is it not an unnecessary risk, respectively, wouldn't that be better in a separate PR?

Comment (vvmruder) [reply pyramid_oereb/core/views/webservice.py:805]: It is related in the context at least. I try to separate the concerns. In this context I avoid passing a pyramid entity to a mehtod which has a really defined interface and therefore I reduce potential errors and rise the testability. I would vote to not split this into another PR because it means the tests need to be redone.

File: tests/core/test_hook_methods.py [hunk 2]
```diff
@@ -60,40 +58,13 @@ def legend_entry_data(pyramid_oereb_test_config, dbsession, transact, file_adapt
     yield legend_entries
 
 
-def test_get_symbol_invalid_theme_code(pyramid_oereb_test_config):
```
Comment (mki-c2c) [inline tests/core/test_hook_methods.py:63]: is there no use for these removed tests ?

**Review by mki-c2c — APPROVED** [review]
Looks good and tests are OK, I have not looked at everything in perfect detail.
Some remarks and questions.

File: pyramid_oereb/contrib/data_sources/standard/hook_methods.py [hunk 3]
```diff
@@ -1,182 +1,75 @@
...
+from pyramid_oereb import database_adapter
 from pyramid_oereb.core import b64
 from pyramid_oereb.core.records.image import ImageRecord
-from pyramid_oereb.core.records.office import OfficeRecord
 
 
 log = logging.getLogger(__name__)
 
 
-def get_symbol(request):
```
Comment (michmuel) [inline pyramid_oereb/contrib/data_sources/standard/hook_methods.py:23]: Regarding https://github.com/openoereb/pyramid_oereb/issues/1476: Is the idea to reorganize the hook_methods in this PR and  to address the symbol issue in a next one?

**Review by michmuel — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 4]
```diff
@@ -80,3 +80,8 @@ DataExtract.json
 /dev/sample_data/ch.real_estate_type.json
 /dev/sample_data/ch.themes.json
 /dev/sample_data/ch.themes_docs.json
+
+/coverage.contrib-data_sources-standard.xml
+/coverage.contrib-print_proxy-mapfish_print.xml
+/coverage.core.xml
+/coverage.xml
```
[the remark this answers is not in the document]
Comment (vvmruder) [reply .gitignore:87]: Yes-I named them according to their content. Or is it still generated?

File: Makefile [hunk 5]
```diff
@@ -254,6 +254,9 @@ build: install $(DEV_CREATE_TABLES_SCRIPT) $(DEV_CONFIGURATION_YML) create_dev_d
 .PHONY: clean
 clean: clean_fed_data clean_dev_db_scripts
 	rm -f $(DEV_CONFIGURATION_YML)
+	rm -f core.xml
```
[the remark this answers is not in the document]
Comment (vvmruder) [reply Makefile:257]: True-I will fix it

File: pyramid_oereb/core/hook_methods.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+from pyramid.path import AssetResolver
+
+from pyramid_oereb import route_prefix
+from pyramid_oereb.core.records.office import OfficeRecord
+
+
+def get_symbol(theme_code, sub_theme_code, view_service_id, type_code, theme_config):
+    """
+    Returns the symbol for the requested theme and type code from database. It queries the model
+    for the legend entry pyramid_oereb.contrib.data_sources.standard.models.get_legend_entry
```
[the remark this answers is not in the document]
Comment (vvmruder) [reply pyramid_oereb/core/hook_methods.py:14]: Yes-better raise not implemented :) you are right.

File: tests/core/test_hook_methods.py (hunk 2 above)
Comment (vvmruder) [reply tests/core/test_hook_methods.py:63]: yes there is but not on Core

File: pyramid_oereb/contrib/data_sources/standard/hook_methods.py (hunk 3 above)
Comment (vvmruder) [reply pyramid_oereb/contrib/data_sources/standard/hook_methods.py:23]: Yes-we will do this later to not get a too big PR.

File: pyramid_oereb/contrib/data_sources/standard/hook_methods.py [hunk 7]
```diff
@@ -1,182 +1,75 @@
...
 
     try:
-        config_parser = StandardThemeConfigParser(**plr)
+
         models = config_parser.get_models()
         model = models.LegendEntry
         legend_entry = session.query(model).filter(
             cast(model.type_code, Text) == cast(type_code, Text)
+        ).filter(
+            model.sub_theme == sub_theme_code
```
[the remark this answers is not in the document]
Comment (vvmruder) [reply pyramid_oereb/contrib/data_sources/standard/hook_methods.py:52]: Even if not really pythonish-sqlalchemy allows `<column>== None` because it translates to SQLish `<column> is NULL` so this code should be save.

File: tests/core/test_hook_methods.py (hunk 2 above)
Comment (vvmruder) [reply tests/core/test_hook_methods.py:63]: Yes there is none-since we do not have any implementation in the core. But only the hook-method dummy.

File: pyramid_oereb/core/hook_methods.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (vvmruder) [reply pyramid_oereb/core/hook_methods.py]: I implemented the method more like a dummy-it now raises the NotImplementedError to avoid direct use.

File: pyramid_oereb/contrib/data_sources/standard/hook_methods.py (hunk 3 above)
Comment (vvmruder) [reply pyramid_oereb/contrib/data_sources/standard/hook_methods.py:23]: @michmuel why are you requesting changes?

File: .gitignore (hunk 4 above)
[the remark this answers is not in the document]
Comment (mki-c2c) [reply .gitignore:87]: I think coverage.xml is not generated any more and can be removed from all mentions in all files

File: pyramid_oereb/contrib/data_sources/standard/hook_methods.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (mki-c2c) [reply pyramid_oereb/contrib/data_sources/standard/hook_methods.py:52]: Yes that's true (== None becomes is NULL) but is this what we want ? I would expect sub_theme == None to match any subtheme ? That was the point of my question. (at least, the behaviour has changed with respect to what was done before)

File: pyramid_oereb/core/hook_methods.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (mki-c2c) [reply pyramid_oereb/core/hook_methods.py]: that's a nice idea (virtual method), thanks

**Comment by mki-c2c** [comment]
could you please also fix codacy findings ?
mako issue can be fixed with exclude paths like here: https://github.com/openoereb/pyramid_oereb/commit/b9c3dd496e3a65b8a6a5cf5bc44596212f32636d
new line issue should be a no-brainer.

File: tests/core/test_hook_methods.py (hunk 2 above)
Comment (mki-c2c) [reply tests/core/test_hook_methods.py:63]: OK. we shoudl just not forget to add them later on in a convenient place.

**Review by michmuel — APPROVED** [review]

File: pyramid_oereb/contrib/data_sources/standard/hook_methods.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (vvmruder) [reply pyramid_oereb/contrib/data_sources/standard/hook_methods.py:52]: `sub_theme==None` matches a theme not a sub_theme
